src/main/java/org/osc/controller/nsc/restserver/api/RestServerApiUtils.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.controller.nsc.restserver.api;
+
+import static java.util.Collections.singletonMap;
+import static org.osgi.service.jdbc.DataSourceFactory.*;
+
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.persistence.EntityManager;
+import javax.sql.DataSource;
+
+import org.osc.controller.nsc.api.SampleSdnRedirectionApi;
+import org.osgi.service.jdbc.DataSourceFactory;
+import org.osgi.service.jpa.EntityManagerFactoryBuilder;
+import org.osgi.service.transaction.control.TransactionControl;
+import org.osgi.service.transaction.control.jpa.JPAEntityManagerProviderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RestServerApiUtils {
+
+    private static Logger LOG = LoggerFactory.getLogger(RestServerApiUtils.class);
+
+    private static ConcurrentHashMap<String, SampleSdnRedirectionApi> sdnApiHashMap = new ConcurrentHashMap<String, SampleSdnRedirectionApi>();
+
+    public static void insertSdnApi(String id, SampleSdnRedirectionApi value) {
+        sdnApiHashMap.putIfAbsent(id, value);
+    }
+
+    public static SampleSdnRedirectionApi getSdnApi(String id) {
+        return sdnApiHashMap.get(id);
+    }
+
+    public static EntityManager createEntityHandle(JPAEntityManagerProviderFactory resourceFactory,
```

Comment:
why do we need to duplicate this code from SampleSdnControllerApi?

can we just inject the SampleSdnControllerApi into the rest api servlet?

Reply:
@arvindn05 , the possibility i see is to have a common method and this method could be called from SampleSdnControllerApi and by the rest layer. This method could be defined in redirectionutil class. Let me know what you think or any other possibility you see? and will wait for your feedback before any change on this front.

Reply:
@arvindn05 , I am waiting for your feedback for submitting this PR.

Reply:
@sudhirappaji can you reply to my original question? Why cant we inject the SampleSdnControllerApi?

Reply:
@arvindn05, yes it should be feasible to inject using Inject annotation. Will confirm after checking it. But need to implement VC passed as a parameter to create method.

Reply:
@arvindn05, I tried injecting SampleSdnControllerApi in the rest layer. This always results in null exception/ as injection is not working. I tried to check/investigate the possible causes, unfortunately did not find any solution. Please guide how to go about? I already spent couple of hours on this topic, but no solution in sight. my suggestion is I can put this as TODO item.